pkg/rpc/server/server.go

```diff
@@ -1847,7 +1862,134 @@ func (s *Server) runScriptInVM(t trigger.Type, script []byte, contractScriptHash
 	if err != nil {
 		faultException = err.Error()
 	}
-	return result.NewInvoke(ic, script, faultException, s.config.MaxIteratorResultItems), nil
+	var registerSession result.OnNewSession
+	if s.config.SessionEnabled {
+		registerSession = s.registerSession
+	}
+	return result.NewInvoke(ic, script, faultException, registerSession), nil
+}
+
+// registerSession is a callback used to add new iterator session to the sessions list.
+// It performs no check whether sessions are enabled.
+func (s *Server) registerSession(sessionID string, iterators map[string]stackitem.Item, finalize func()) {
+	s.sessionsLock.Lock()
+	defer s.sessionsLock.Unlock()
+
+	sess := &session{
+		Iterators: iterators,
+		Till:      time.Now().Add(time.Second * time.Duration(s.config.SessionExpirationTime)),
+		Finalize:  finalize,
+	}
+	s.sessions[sessionID] = sess
+}
+
+func (s *Server) handleSessions() {
+	if !s.config.SessionEnabled {
+		return
+	}
+	s.log.Info("rpc-server iterator sessions are enabled")
+	var (
+		finalizationInterval = time.Duration(s.config.SessionExpirationTime) * time.Second
+		finalizationTicker   = time.NewTicker(finalizationInterval)
+	)
+
+	defer func() {
+		finalizationTicker.Stop()
+		s.sessionsLock.Lock()
+		defer s.sessionsLock.Unlock()
+		for _, session := range s.sessions {
+			session.Finalize()
+		}
+		s.sessions = nil
+	}()
+
+	for {
+		select {
+		case <-s.shutdown:
+			return
+		case <-finalizationTicker.C:
+			s.sessionsLock.Lock()
+			if len(s.sessions) == 0 {
+				s.sessionsLock.Unlock()
+				continue
+			}
+			now := time.Now()
+			staleSessions := s.sessions
+			s.sessions = make(map[string]*session)
+			for id, sess := range staleSessions {
+				if sess.Till.Before(now) {
+					sess.Finalize()
+				} else {
+					s.sessions[id] = sess
+				}
+			}
```

Comment:
Because the ticker has tick period equal to the session expiration, so likely, the whole set of sessions will be removed from map at the subsequent tick. Isn't it cheaper than using `delete` for all set of elements?

Reply:
Could you add this as a comment?

Reply:
Added.